libbeat/outputs/logstash/async.go

```diff
@@ -197,6 +198,9 @@ func (c *asyncClient) publishWindowed(
 }
 
 func (c *asyncClient) sendEvents(ref *msgRef, events []publisher.Event) error {
+	if c.client == nil {
+		return errors.New("connection closed")
+	}
```

Comment:
The `Close` method can be called from another thread. Better use an atomic pointer or mutex to load `c.client` into a local variable. The `v2.AsyncClient` should be thread safe (just uses `net.Conn.Close()` to close the underlying connection/FD).

Reply:
Fixed.